components/brave_wallet/browser/internal/orchard_block_scanner.cc

```diff
@@ -29,50 +33,57 @@ OrchardBlockScanner::~OrchardBlockScanner() = default;
 
 base::expected<OrchardBlockScanner::Result, OrchardBlockScanner::ErrorCode>
 OrchardBlockScanner::ScanBlocks(
-    std::vector<OrchardNote> known_notes,
+    OrchardTreeState tree_state,
     std::vector<zcash::mojom::CompactBlockPtr> blocks) {
+  std::unique_ptr<orchard::OrchardDecodedBlocksBundle> result =
+      decoder_->ScanBlocks(tree_state, blocks);
+  if (!result) {
+    DVLOG(1) << "Failed to parse block range.";
+    return base::unexpected(ErrorCode::kInputError);
+  }
+
+  if (!result->GetDiscoveredNotes()) {
```

Comment:
This function is declared:

```cxx
  std::optional<std::vector<::brave_wallet::OrchardNote>> GetDiscoveredNotes()
```

This means that each call to this function results in a vector being constructed, which may include allocations. To avoid allocating this vector twice, move the declaration of `found_notes` to this point, and have `found_notes` to be optional vector.